### PR TITLE
docs: add large segments explainer

### PR DESCRIPTION
**Pull request by thomasheartman**
This PR adds an explainer document about using large/complex constraints, inspired by the Unleash and Redis repo.

The intent is to create a comprehensive and understandable explanation as to why we don't recommend you do this, and also to offer alternative solutions for the same problem.

### CODE REVIEW

File: website/docs/topics/large-segments.mdx [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+Because of their availability and ease of use with smaller lists, it can be tempting to just keep adding identifiers to those lists. However, once you start approaching a hundred elements we recommend that you find another way to manage these IDs. In fact, it's probably better to stop well before you get that far.
+
+The rest of this document will explain why this is generally a bad idea and how we suggest you solve your problems instead. However, as always there **are** exceptional cases where this is the solution.
+
+How large is too large? How complex is too complex? It depends. However, smaller and simpler is generally better.
+
+## The cost of data
+
+First, let's talk a bit about Unleash's architecture: **Your Unleash instance**is where you define all your features, their strategies, and constraints (and a lot more). However, the Unleash instance itself does not do any of the feature evaluation[^1]. That responsibility is relegated to the Unleash SDKs (or [Edge](/generated/unleash-edge.md) / the [Unleash proxy](/generated/unleash-proxy.md)).
```
Comment (sighphyre) [inline website/docs/topics/large-segments.mdx:23]: Delegated maybe? It's an important responsibility. Entirely optional suggestion

```suggestion
First, let's talk a bit about Unleash's architecture: **Your Unleash instance**is where you define all your features, their strategies, and constraints (and a lot more). However, the Unleash instance itself does not do any of the feature evaluation[^1]. That responsibility is delegated to the Unleash SDKs (or [Edge](/generated/unleash-edge.md) / the [Unleash proxy](/generated/unleash-proxy.md)).
```

**Review by sighphyre — APPROVED** [review]
Yes! This is fantastic!

File: website/docs/topics/large-segments.mdx (hunk 1 above)
Comment (chriswk) [reply website/docs/topics/large-segments.mdx:23]: +1 for delegated here

Comment (thomasheartman) [reply website/docs/topics/large-segments.mdx]: Yeah, I'm good with delegated. Good suggestion 🙌🏼

**Pull request merged**